runtime/lua/vim/lsp/util.lua

```diff
@@ -575,6 +575,9 @@ end
 --- precedence is as follows: textEdit.newText > insertText > label
 --see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion
 local function get_completion_word(item)
+  if item.filterText ~= nil then
+    return item.filterText
+  end
```

Comment:
I'm afraid this doesn't match the specification. `filterText` should only be used for filtering and not be used as item that is inserted when selected.

Is this really required to fix your issue?

Reply:
You're right, it's not needed